payments/payment_manager.go

```diff
@@ -0,0 +1,95 @@
+package payments
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/types"
+)
+
+type (
+	// Balance stores the remaining and paid funds in a channel.
+	Balance struct {
+		Remaining *big.Int
+		Paid      *big.Int
+	}
+
+	// PaymentManager can be used to make a payment for a given channel, issuing a new, signed voucher to be sent to the receiver
+	PaymentManager struct {
+		signer   common.Address
+		channels map[types.Destination]*Balance
+	}
```

Comment:
Could this struct be private (non exported)? Perhaps an interface should be exported, instead of a concrete type.

Reply:
What's the benefit of exporting an interface over a concrete type?

Reply:
My thinking was that not exporting the concrete type forces consumers to use the constructor, rather than doing things manually. Nice to have in general, but perhaps not that important here. 

If we were not exporting the concrete type, it might be useful for there to be an interface exported, so that consumers know how to interact with e.g. a `PaymentManager`. Probably also not super important right now.

Reply:
What you're saying makes sense, but I understood that to be an anti-pattern in the go ecosystem -- that we are encouraged to use interfaces where types are _used_, not where types are _defined_.  That said, I'm struggling to find a reference/reason for **not** defining interfaces where we define types.

It's easy to see why we'd be encouraged to use interfaces to define what we need out of a function argument. It also seems commonplace (and fine) to allow people to construct structs explicitly, especially when the zero value is useful! (Unfortunately, the zero value is not useful here.)

@NiloCK do you remember what I'm thinking of? Unless I can be convinced otherwise, I'm happy to export an interface.

Reply:
73deee843ade73762c5090c9df9e550b1bfc589e